src/components/jacobian_component/jacobian.sh

```diff
@@ -151,52 +185,176 @@ create_simulation_dir() {
-            PerturbBCValues=$(generate_BC_perturb_values $bcThreshold $x $PerturbValueBCs)
+
+        # If the current state vector element is one of the BC state vector elements, then
+        # turn on BC optimization for the corresponding edge
+        if [[ $start_element -gt $bcThreshold ]] && [[ "$OH_elem" = false ]]; then
+            BC_elem=true
+            PerturbBCValues=$(generate_BC_perturb_values $bcThreshold $start_element $PerturbValueBCs)
             sed -i -e "s|CH4_boundary_condition_ppb_increase_NSEW:.*|CH4_boundary_condition_ppb_increase_NSEW: ${PerturbBCValues}|g" \
-                -e "s|perturb_CH4_boundary_conditions: false|perturb_CH4_boundary_conditions: true|g" \
-                -e "s|emission_perturbation_factor: ${PerturbValue}|emission_perturbation_factor: 1.0|g" \
-                -e "s|state_vector_element_number: ${xUSE}|state_vector_element_number: 0|g" geoschem_config.yml
+                -e "s|perturb_CH4_boundary_conditions: false|perturb_CH4_boundary_conditions: true|g" geoschem_config.yml
         fi
     fi
 
-    if "$OptimizeOH"; then
-        # The last state vector element is reserved for OH optimization.
-        # If this is the current state vector element, then modify the OH
-        # perturb value in HEMCO_Config.rc and revert emission perturbation.
-        OHthreshold=$(($nElements - 1))
-        if [ $x -gt $OHthreshold ]; then
-            sed -i -e "s|emission_perturbation: ${PerturbValue}|emission_perturbation: 1.0|g" \
-                -e "s|state_vector_element_number: ${xUSE}|state_vector_element_number: 0|g" geoschem_config.yml
-            sed -i -e "s| OH_pert_factor  1.0| OH_pert_factor  ${PerturbValueOH}|g" HEMCO_Config.rc
+    # the prior, OH perturbation, and background simulations need to have soil absorption
+    # and, in the case, of kalman mode the prior is scaled by the nudged scale factors
+    if [[ $x -eq 0 ]] || [[ "$x" = "background" ]] || [[ $OH_elem = true ]]; then
+        # Use MeMo soil absorption for the prior simulation
+        sed -i -e "/(((MeMo_SOIL_ABSORPTION/i ))).not.UseTotalPriorEmis" \
+            -e "/)))MeMo_SOIL_ABSORPTION/a (((.not.UseTotalPriorEmis" HEMCO_Config.rc
+        if "$KalmanMode"; then
+            # Use nudged scale factors for the prior simulation and OH simulation for kalman mode
+            sed -i -e "s|--> Emis_PosteriorSF       :       false|--> Emis_PosteriorSF       :       true|g" \
+                -e "s|--> UseTotalPriorEmis      :       false|--> UseTotalPriorEmis      :       true|g" \
+                -e "s|gridded_posterior.nc|${RunDirs}/ScaleFactors.nc|g" HEMCO_Config.rc
+        fi
+
+    else
+        # set 1ppb CH4 boundary conditions and restarts for all other perturbation simulations
+        # Note that we use the timecycle flag C to avoid having to make additional files
+        RestartFile=${RunDirs}/jacobian_1ppb_ics_bcs/Restarts/GEOSChem.Restart.1ppb.${StartDate}_0000z.nc4
```

Comment:
If using BCs for the Restart (which I am doing for my global simulation), then the Restart File in `jacobian_1ppb_ics_bcs/Restarts/` is called `GEOSChem.**BoundaryConditions**.1ppb.${StartDate}_0000z.nc4` instead of `GEOSChem.**Restart**.1ppb.${StartDate}_0000z.nc4`, so I added logic to prevent a broken symlink from being created 10 lines down, like this:
[https://github.com/megan-he/integrated_methane_inversion/commit/52aebb0721edf9b036f1ee848da09c05517894a5](url)

Reply:
@megan-he I am guessing your fork of the IMI is private, so I cant see your fix :(

Reply:
Here is the correct link https://github.com/megan-he/integrated_methane_inversion/commit/52aebb0721edf9b036f1ee848da09c05517894a5